internal/store/installation.go

```diff
@@ -199,7 +204,39 @@ func (sqlStore *SQLStore) GetUnlockedInstallationsPendingWork() ([]*model.Instal
 }
 
 // CreateInstallation records the given installation to the database, assigning it a unique ID.
-func (sqlStore *SQLStore) CreateInstallation(installation *model.Installation) error {
+func (sqlStore *SQLStore) CreateInstallation(installation *model.Installation, annotations []*model.Annotation) error {
```

Comment:
Could you help me understand the value that the DTO object brings for holding Annotations versus just storing them directly on the `model.Thing` (`model.Installation` in this case) object directly? Using the new DTO objects to hold this extra metadata requires a great number of changes to call sites and return types, seen throughout this change, including here.

Reply:
As annotations are stored in a different table than the Installation/Cluster I do not think it is good to store them directly in the struct.
For example, if I only need the Installation object I do not need to join other tables containing annotations data and at the same, I avoid confusion caused by pulling only the Installation object with `Annotations` field being `nil` when in fact there might be some annotations assign for this particular Installation.

Reply:
Ok I had a suspicion after thinking about this a bit that it had to do with the underlying data model. Thank you for the explanation